x-pack/solutions/security/plugins/security_solution/docs/adrs/threat_hunting/0002_timeline_ui_batching_strategy.md

```diff
@@ -0,0 +1,172 @@
+# Timeline Batching Functional Design ADR
+
+
+
+This document details the design of how Timeline Batching is supposed to work. The purpose of this document is to have a source of truth where reviewers of any changes can refer back and decide if the change is in line with the design or not.
+
+This document is only applicable for Timeline datagrid.
+
+## What is a batch
+Batch ( also known as Sample ) in Unified Data table is the collection of max number of documents that are fetched in a single request. When a user queries a timeline for the first time, Batch 0 is fetched and then the user may have the option to fetch further batches on-demand.
+
+
+### Batch v/s Data Grid Pagination
+The concept of batch is different from the Datagrid Pagination. Datagrid Pagination is completely client driven and datagrid may decide how it wants to paginate a particular batch.
```

Comment:
very minor, but I think it'd be best to stick with one word when we talk about the table. Above we use `table`, here we use `Datagrid` and `datagrid`.

Reply:
changed all instanced to `Datagrid` in fd2ababe4e4